// router/users.js
const express = require("express");
const bcrypt = require("bcrypt");
const mysql = require("mysql");
const router = express.Router();

const connection = mysql.createConnection({
  host: "localhost",
  user: "root",
  password: "root",
  database: "redlucia"
});

// Ruta para registrar un nuevo usuario
router.post("/register", (req, res) => {
  const { nombre, apellidos, telefono, contrasena, tipo_usuario } = req.body;

  // El usuario será igual al número de teléfono
  const usuario = telefono;
  const dispositivos_maximos = tipo_usuario === "Dedicado" ? 5 : 1;

  bcrypt.hash(contrasena, 10, (err, hash) => {
    if (err) {
      return res.status(500).json({
        success: false,
        mensaje: "Error al cifrar contraseña"
      });
    }

    const sql = `
      INSERT INTO usuarios 
      (nombre, apellidos, telefono, usuario, contrasena, tipo_usuario, dispositivos_maximos) 
      VALUES (?, ?, ?, ?, ?, ?, ?)
    `;

    const valores = [
      nombre,
      apellidos,
      telefono,
      usuario,
      hash,
      tipo_usuario,
      dispositivos_maximos
    ];

    connection.query(sql, valores, (err) => {
      if (err) {
        if (err.code === 'ER_DUP_ENTRY') {
          return res.json({
            success: false,
            mensaje: "Usuario o teléfono ya registrado"
          });
        }
        return res.status(500).json({
          success: false,
          mensaje: "Error al registrar"
        });
      }

      res.json({
        success: true,
        mensaje: "Usuario registrado correctamente"
      });
    });
  });
});

module.exports = router;
